src/Http/Client/Adapter/Curl.php

```diff
@@ -148,11 +148,15 @@ protected function getProtocolVersion(Request $request)
                 return CURL_HTTP_VERSION_1_0;
             case '1.1':
                 return CURL_HTTP_VERSION_1_1;
+            case '2':
             case '2.0':
+                if (defined('CURL_HTTP_VERSION_2TLS')) {
```

Comment:
I wish we could warn that it is falling back to non-tls.  It might make a connection and they think tls is configured right.

Reply:
Maybe can update the minimum for 4.x

Reply:
HTTP/2 without TLS is very rare, and if this constant is used the request should fail based on the documentation.

Reply:
HTTP/2 without TLS is very rare, and if this constant is used the request should fail based on the documentation.

Reply:
I was talking about the constant not being defined.

Reply:
We would need to see what version of curl is in the LTS of major distros before we could bump our requirement up. Curl is also an optional dep of the client so we could only enforce the dependency at runtime. We could document the suggested versions though.